sqlalchemy_bigquery/base.py

```diff
@@ -56,7 +56,9 @@
 from .parse_url import parse_url
 from . import _helpers, _struct, _types
 
-FIELD_ILLEGAL_CHARACTERS = re.compile(r"[^\w]+")
+# Incorporates special characters used in flexible column names:
+# https://cloud.google.com/bigquery/docs/schemas#flexible-column-names
+FIELD_ILLEGAL_CHARACTERS = re.compile(r"[^\w&%=+:'<>#|]+", re.ASCII)
```

Comment:
Whitespace is allowed now, right? I used `[^!"$()*,./;?@[\\\]^`{}~\n]*` in https://github.com/ibis-project/ibis/pull/6557 and it seems to be working alright.